Compilation warnings

```
[44/216] Building C object src/CMakeFiles/supplements.dir/seis/pscoupe.c.o
../../src/seis/pscoupe.c:530:62: warning: unused parameter 'GMT' [-Wunused-parameter]
GMT_LOCAL unsigned int pscoupe_parse_old_A (struct GMT_CTRL *GMT, struct PSCOUPE_CTRL *Ctrl, char *arg) {
                                                             ^
../../src/seis/pscoupe.c:561:21: warning: unused variable 'lon2' [-Wunused-variable]
        double lon1, lat1, lon2, lat2;
                           ^
../../src/seis/pscoupe.c:561:27: warning: unused variable 'lat2' [-Wunused-variable]
        double lon1, lat1, lon2, lat2;
                                 ^
../../src/seis/pscoupe.c:561:9: warning: unused variable 'lon1' [-Wunused-variable]
        double lon1, lat1, lon2, lat2;
               ^
../../src/seis/pscoupe.c:561:15: warning: unused variable 'lat1' [-Wunused-variable]
        double lon1, lat1, lon2, lat2;
                     ^
5 warnings generated.
[94/216] Building C object src/CMakeFiles/gmtlib.dir/gmt_stat.c.o
../../src/gmt_stat.c:2842:14: warning: unused variable 'm' [-Wunused-variable]
        uint64_t k, m = 0;
                    ^
1 warning generated.
[133/216] Building C object src/CMakeFiles/gmtlib.dir/gmt_init.c.o
../../src/gmt_init.c:6440:6: warning: unused variable 'i' [-Wunused-variable]
        int i, error = 0;
            ^
1 warning generated.
[136/216] Building C object src/CMakeFiles/gmtlib.dir/gmt_support.c.o
../../src/gmt_support.c:6965:38: warning: unused parameter 'GMT' [-Wunused-parameter]
void gmt_scale_pen (struct GMT_CTRL *GMT, struct GMT_PEN *P, double scale) {
                                     ^
1 warning generated.
[182/216] Building C object src/CMakeFiles/gmtlib.dir/psevents.c.o
../../src/psevents.c:496:16: warning: unused variable 'was' [-Wunused-variable]
                                                char *q, was;
                                                         ^
../../src/psevents.c:589:69: warning: unused parameter 'GMT' [-Wunused-parameter]
GMT_LOCAL unsigned int psevents_determine_columns (struct GMT_CTRL *GMT, char *module, char *cmd, unsigned int mode) {
                                                                    ^
2 warnings generated.
[186/216] Building C object src/CMakeFiles/gmtlib.dir/grdmath.c.o
../../src/grdmath.c:2401:9: warning: unused variable 'F' [-Wunused-variable]
        double F, lon, lat, kappa;
               ^
1 warning generated.
```

--- a/src/gmt_init.c
+++ b/src/gmt_init.c
@@ -6437,7 +6437,7 @@ GMT_LOCAL void gmtinit_conf_classic (struct GMT_CTRL *GMT) {
 
 /*! . */
 GMT_LOCAL void gmtinit_conf_modern_override (struct GMT_CTRL *GMT) {
-	int i, error = 0;
+	int error = 0;
 #if NO_THEMES
 	return;		/* Ignore all the modern theme stuff */
 #endif

--- a/src/gmt_stat.c
+++ b/src/gmt_stat.c
@@ -2839,7 +2839,7 @@ void gmt_get_cellarea (struct GMT_CTRL *GMT, struct GMT_GRID *G) {
 double gmt_von_mises_mu_and_kappa (struct GMT_CTRL *GMT, double *data, double *w, uint64_t n, double *kappa) {
 	/* Return the mean and kappa for a von Mises fit to (possibly weighted) data.
 	 * It is assumed that data have been scaled to 0-360. Weights w is possibly NULL for no weights */
-	uint64_t k, m = 0;
+	uint64_t k;
 	double mean = 0.0, x_r = 0.0, y_r = 0.0, s_w = 0.0, ww = 1.0;
 	double lo, hi, midval, range2, delta_R, s, c, R_bar;
 

--- a/src/gmt_support.c
+++ b/src/gmt_support.c
@@ -6964,6 +6964,7 @@ void gmt_freepen (struct GMT_CTRL *GMT, struct GMT_PEN *P) {
 
 void gmt_scale_pen (struct GMT_CTRL *GMT, struct GMT_PEN *P, double scale) {
 	/* Scale all pen attributes by given scale. Note: P as assumed to be reset to nominal values before scaling */
+	gmt_M_unused (GMT);
 	P->width  *= scale;
 	P->offset *= scale;
 	if (P->style[0]) {	/* Must scale the dashes and gaps and update string */

--- a/src/grdmath.c
+++ b/src/grdmath.c
@@ -2398,7 +2398,7 @@ GMT_LOCAL void grdmath_FISHER (struct GMT_CTRL *GMT, struct GRDMATH_INFO *info, 
 {
 	uint64_t node;
 	unsigned int prev1, prev2, row, col;
-	double F, lon, lat, kappa;
+	double lon, lat, kappa;
 
 	prev1 = last - 1;
 	prev2 = last - 2;

--- a/src/psevents.c
+++ b/src/psevents.c
@@ -493,7 +493,7 @@ static int parse (struct GMT_CTRL *GMT, struct PSEVENTS_CTRL *Ctrl, struct GMT_O
 				Ctrl->Z.active = true;
 				if (opt->arg[0] && strstr (opt->arg, "-S")) {	/* Got the required -S option as part of the command */
 					if ((c = strchr (opt->arg, ' '))) {	/* First space in the command ends the module name */
-						char *q, was;
+						char *q;
 						c[0] = '\0';	/* Temporarily hide the rest of the command so we can isolate the module name */
 						Ctrl->Z.module = strdup (opt->arg);	/* Make a copy of the module name */
 						c[0] = ' ';	/* Restore space */
@@ -590,6 +590,7 @@ GMT_LOCAL unsigned int psevents_determine_columns (struct GMT_CTRL *GMT, char *m
 	/* Return how many data columns are needed for the selected seismo/geodetic symbol */
 	unsigned int n = 0;
 	char *S = strstr (cmd, "-S");	/* Pointer to start of symbol option */
+	gmt_M_unused (GMT);
 	S += 2;	/* Now at format designation */
 
 	if (strstr (module, &coupe[2])) {	/* Using coupe/pscoupe */

--- a/src/seis/pscoupe.c
+++ b/src/seis/pscoupe.c
@@ -558,7 +558,6 @@ static int parse (struct GMT_CTRL *GMT, struct PSCOUPE_CTRL *Ctrl, struct GMT_OP
 	unsigned int n_errors = 0;
 	char txt_a[GMT_LEN256] = {""}, txt_b[GMT_LEN256] = {""}, txt_c[GMT_LEN256] = {""}, txt_d[GMT_LEN256] = {""}, *p = NULL;
 	struct GMT_OPTION *opt = NULL;
-	double lon1, lat1, lon2, lat2;
 
 	for (opt = options; opt; opt = opt->next) {	/* Process all the options given */
 

